### PR TITLE
Add branch options

### PR DESCRIPTION
**Pull request by rwedge**
- Add option to specify branch to run against for Tasks (is_workflow_success, project_build, is_recent_commit)
- Add instructions to readme
- Add to `main.py` to allow Github actions to pass `branch` argument, defaults to None (which uses default repo branch)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,5 +12,5 @@ lint-tests:
 	isort --check-only --recursive circleci
 
 unit-tests:
-	pytest circleci/test.py --cov=circleci --circle-token ${CIRCLE_TOKEN} \
+	pytest -s circleci/test.py --cov=circleci --circle-token ${CIRCLE_TOKEN} \
```
Comment (gsheni) [inline Makefile:15]: Maybe leave in for future debugging?

File: circleci/api.py [hunk 2]
```diff
@@ -1,46 +1,67 @@
...
-def latest_workflow(repository, circle_token='', status='completed'):
-    url = CIRCLE_API + "/project/github/{0}/tree"
-    url += '/master?circle-token={1}&filter={2}'
-    url = url.format(repository, circle_token, status)
+def latest_workflow(repository, circle_token="", status="completed",
+                    branch=None):
+    if branch is None:
+        branch = default_branch(repository)
+    # CircleCI API requires url-encoded branch
+    branch = urllib.parse.quote(branch)
```
Comment (gsheni) [inline circleci/api.py:26]: See here:
https://circleci.com/docs/api/#recent-builds-for-a-single-project


> Note: You can narrow the builds to a single branch by appending /tree/:branch to the url. Note that the branch name should be url-encoded.

File: .circleci/config.yml [hunk 3]
```diff
@@ -3,6 +3,6 @@ version: 2.1
 jobs:
   build:
     docker:
-      - image: circleci/purposeful_failure:2.4.1
+      - image: circleci/ruby:2.4.1
```
Comment (rwedge) [inline .circleci/config.yml:6]: since we only need `echo`, how about `busybox` for the image

File: circleci/api.py [hunk 4]
```diff
@@ -1,46 +1,67 @@
...
-    url = CIRCLE_API + "/project/github/{0}/build?circle-token={1}"
-    response = requests.post(url.format(repository, circle_token))
-    info = '%s (%s)' % (response.reason, response.status_code)
+def project_build(repository, circle_token="", branch=None):
+    if branch is None:
+        branch = default_branch(repository)
+        # CircleCI API requires url-encoded branch
+        branch = urllib.parse.quote(branch)
+    url = CIRCLE_API
+    url += f"/project/github/{repository}/build?circle-token={circle_token}"
```
Comment (gsheni) [inline circleci/api.py:52]: This URL endpoint does seem to require an url-encoded branch (according to docs)
https://circleci.com/docs/api/#trigger-a-new-build-by-project-preview

File: .circleci/config.yml (hunk 3 above)
Comment (gsheni) [reply .circleci/config.yml]: fixed.

File: README.md [hunk 5]
```diff
@@ -27,12 +27,13 @@ This is a list of the available tasks:
...
 
-| Parameters   | Required | Description                                              |
-|--------------|----------|----------------------------------------------------------|
-| repository   | yes      | The repository to check for a sucessful workflow status. |
-| circle-token | yes      | A personal API token to access the CircleCI API.         |
+| Parameters   | Required | Description                                                                            |
+|--------------|----------|----------------------------------------------------------------------------------------|
+| repository   | yes      | The repository to check for a sucessful workflow status.                               |
+| circle-token | yes      | A personal API token to access the CircleCI API.                                       |
+| branch       | no       | The branch to check aganist. If none is specified, uses default branch for repository. |
```
Comment (rwedge) [inline README.md:36]: > The branch to check aganist

typo in "against".  same typo in the other places where branch was added as a parameter

Comment (gsheni) [reply README.md]: Fixed

File: circleci/api.py (hunk 2 above)
Comment (rwedge) [reply circleci/api.py:26]: Are we assuming the branch name we get back from github is url encoded?

Comment (gsheni) [reply circleci/api.py:26]: Unsure on this. The API is not clear about this.
https://developer.github.com/v3/repos/#get-a-repository

File: circleci/api.py (hunk 4 above)
Comment (rwedge) [reply circleci/api.py:52]: should we do encoding here like we do in latest_workflow?

Comment (gsheni) [reply circleci/api.py:54]: Added. Though I am not sure if that endpoint will require it, because the docs aren't clear about it.

**Pull request merged**